csharp-architectural-patterns/BrokerPattern/MessagingService/MessageService.cs

```diff
@@ -0,0 +1,116 @@
+﻿using Azure.Messaging.ServiceBus;
+
+namespace MessageService
+{
+    public class MessageService : IMessageService
+    {
+        Action<string>? messageHandlerCallback;
+        private static readonly string connectionString = "{service bus connection string}";
+
+        public async Task SendMessageAsync(string queueOrTopicName, string message)
+        {
+            ServiceBusClient client = CreateClient();
+
+            ServiceBusSender? sender = client.CreateSender(queueOrTopicName);
+            if (sender is null)
+            {
+                return;
+            }
+
+            try
+            {
+                ServiceBusMessage serviceBusMessage = new(message);
+                await sender.SendMessageAsync(serviceBusMessage);
+            }
+            finally
+            {
+                await sender.DisposeAsync();
+                await client.DisposeAsync();
+            }
+        }
+
+        public async Task ReceiveMessagesFromQueueAsync(string queueName, Action<string> callback, int millisecondsDelay)
+        {
+            messageHandlerCallback = callback;
+            ServiceBusClient client = CreateClient();
+
+            if (client is null)
+            {
+                return;
+            }
+
+            ServiceBusProcessor? processor = client.CreateProcessor(queueName, new ServiceBusProcessorOptions());
+
+            try
+            {
+                await ProcessMessages(millisecondsDelay, processor);
+            }
+            finally
+            {
+                await processor.DisposeAsync();
+                await client.DisposeAsync();
+            }
+        }
+
+        public async Task ReceiveMessagesWithSubscriptionAsync(string topicName, string subscriptionName, Action<string> callback, int millisecondsDelay)
+        {
+            messageHandlerCallback = callback;
+            ServiceBusClient client = CreateClient();
+
+            if (client is null)
+            {
+                return;
+            }
+
+            ServiceBusProcessor? processor = client.CreateProcessor(topicName, subscriptionName, new ServiceBusProcessorOptions());
+
+            try
+            {
+                await ProcessMessages(millisecondsDelay, processor);
+            }
+            finally
+            {
+                await processor.DisposeAsync();
+                await client.DisposeAsync();
+            }
+
+        }
+
+        private static ServiceBusClient CreateClient()
+        {
+            var clientOptions = new ServiceBusClientOptions()
+            {
+                TransportType = ServiceBusTransportType.AmqpWebSockets
+            };
+            ServiceBusClient? client = new(connectionString, clientOptions);
+            return client;
```

Comment:
The `return` statement should always have a blank line before it. Make sure to apply it where possible.